falcon/asgi/ws.py

```diff
@@ -529,6 +551,25 @@ def _translate_webserver_error(self, ex: Exception) -> Optional[Exception]:
                 'WebSocket subprotocol must be from the list sent by the client'
             )
 
+        # NOTE(vytas): Per ASGI HTTP & WebSocket spec v2.4:
+        #   If send() is called on a closed connection the server should raise
+        #   a server-specific subclass of IOError.
+        # NOTE(vytas): Uvicorn 0.30.6 seems to conform to the spec only when
+        #   using the wsproto stack, it then raises an instance of
+        #   uvicorn.protocols.utils.ClientDisconnected.
+        if isinstance(ex, OSError):
+            close_code = None
+
+            # NOTE(vytas): If using the "websockets" backend, Uvicorn raises
+            #   and instance of OSError from a websockets exception like this:
+            #   "received 1001 (going away); then sent 1001 (going away)"
+            if ex.__cause__:
```

Comment:
I would try to be less strict here.
Maybe something like:
```
msg = str(ex.__cause__ or ex)
match = _CLIENT_DISCONNECTED_CAUSE.match(str(ex.__cause__))
...
```

we can assume that this may match a wider range of servers

Reply:
Hm, good point, but OTOH from what I have been testing other messages looked very different. Also the existing code for Daphne/Autobahn is very specific, so maybe we could start it like this.